### ISSUE TITLE
Subgraph bug

### ISSUE THREAD

**Issue by guyemerson** [delph-in/pydelphin#40]
If the list of nodeids includes a node qeq to the ltop, then I get the following error:

  File "...\delphin\mrs\xmrs.py", line 413, in subgraph
    lnk=self.lnk, surface=self.surface, identifier=self.identifier
  File "...\delphin\mrs\xmrs.py", line 87, in __init__
    self.add_hcons(hcons)
  File "...\delphin\mrs\xmrs.py", line 146, in add_hcons
    'Handle constraint already exists for hole %s.' % hi
delphin._exceptions.XmrsError: Handle constraint already exists for hole h0.

For example, if there's an hcon "h0 qeq h1", and there's an ep with the label h1 and the nodeid 10001, and you call subgraph([10001]), I get this error.

**Comment by goodmami** [comment]
Sorry for letting this one go for a while. I'm unable to reproduce, though:

```python
>>> s
'[ LTOP: h0 INDEX: e2 [ e SF: prop TENSE: pres MOOD: indicative PROG: - PERF: - ] RELS: < [ udef_q_rel<0:10> LBL: h4 ARG0: x3 [ x PERS: 3 NUM: pl IND: + ] RSTR: h5 BODY: h6 ]  [ "_large_a_1_rel"<0:5> LBL: h7 ARG0: e8 [ e SF: prop TENSE: untensed MOOD: indicative ] ARG1: x3 ]  [ "_dog_n_1_rel"<6:10> LBL: h7 ARG0: x3 ]  [ "_bark_v_1_rel"<11:15> LBL: h1 ARG0: e2 ARG1: x3 ]  [ "_loud_a_1_rel"<16:23> LBL: h1 ARG0: e9 [ e SF: prop TENSE: untensed MOOD: indicative PROG: - PERF: - ] ARG1: e2 ] > HCONS: < h0 qeq h1 h5 qeq h7 > ]'
>>> m = simplemrs.loads_one(s)
>>> m.ep(10003)
<ElementaryPredication object ("_bark_v_1_rel" (e2)) at 139640359245168>
>>> m.top
'h0'
>>> m.hcon(m.top)
('h0', 'qeq', 'h1')
>>> m.hcon(m.top)[2] == m.ep(10003).label
True
>>> print(simpledmrs.dumps_one(m.subgraph([10003]), pretty_print=True))
dmrs {
  10003 ["_bark_v_1_rel"<11:15> e PROG=- SF=prop TENSE=pres PERF=- MOOD=indicative];
  0:/H -> 10003;
}
```

Do you still have the *MRS that caused the error? There might be something else special about it.

**Comment by guyemerson** [comment]
The test_subgraph method in mrs_xmrs_test.py has an example that failed.

Perhaps there's a difference when initialising from DMRS nodes and links, compared to initialising from SimpleMRS.

**Comment by goodmami** [comment]
Oh, yes, I think you created the test (df079fb03c6c4bf4e1d7fad6fdb18bace3f3f83f) and the fix (3fc3592bbcef19606e4dbffcdef18c6594ba4f4d) a while ago. The commit message didn't have a "Fix #40" message nor did this issue get updated, so I didn't realize it was already done. I'll label this as "FixCommitted", and it should get closed when that commit is merged to master.

Thanks!
